src/test/java/spoon/test/record/CtRecordTest.java

```diff
@@ -146,6 +146,16 @@ private CtModel createModelFromPath(String code) {
 		return model;
 	}
 
+	@Test
+	void testGenericTypeParametersArePrintedBeforeTheFunctionParameters() {
+		// a record with generic type arguments should be printed correctly 
```

Comment:
Could you add a `// contract:  ... a record with generic type arguments should be printed correctly`?

Reply:
done